IoTuring/Configurator/Configurator.py

```diff
@@ -305,96 +341,48 @@ def ShowUnsupportedEntities(self, ecm: EntityClassManager):
-
-            self.EntityMenuPresetToConfiguration(entityName, preset)
-        except UserCancelledException:
-            self.DisplayMessage("Configuration cancelled", force_clear=True)
-
-        except Exception as e:
-            print("Error during entity preset loading: " + str(e))
-
-        self.ManageEntities()
-
-    def IsEntityActive(self, entityName) -> bool:
-        """ Return True if an Entity is active """
-        for entity in self.config[KEY_ACTIVE_ENTITIES]:
-            if entityName == entity[KEY_ENTITY_TYPE]:
-                return True
-        return False
-
-    def GetEntityLabel(self, entityConfig) -> str:
-        """ Get the type name of entity, add tag if multi"""
-        entityLabel = entityConfig[KEY_ENTITY_TYPE]
-        if KEY_ENTITY_TAG in entityConfig:
-            entityLabel += f" with tag {entityConfig[KEY_ENTITY_TAG]}"
-        return entityLabel
-
-    def RemoveActiveEntity(self, entityConfig) -> None:
-        """ Remove entity name from the list of active entities if present """
-        if entityConfig in self.config[KEY_ACTIVE_ENTITIES]:
-            self.config[KEY_ACTIVE_ENTITIES].remove(entityConfig)
-
+    def RemoveActiveConfiguration(self, singleConfig: SingleConfiguration) -> None:
+        """ Remove configuration (wh or entity) """
+        self.config.RemoveActiveConfiguration(singleConfig)
         self.DisplayMessage(
-            f"Entity removed: {self.GetEntityLabel(entityConfig)}")
-        self.ManageEntities()
+            f"{singleConfig.GetCategoryName()} removed: {singleConfig.GetLabel()}")
 
-    def IsWarehouseActive(self, warehouseName) -> bool:
+    def IsWarehouseActive(self, whClass) -> bool:
         """ Return True if a warehouse is active """
-        for wh in self.config[KEY_ACTIVE_WAREHOUSES]:
-            if warehouseName == wh[KEY_WAREHOUSE_TYPE]:
-                return True
-        return False
+        return bool(self.config.GetConfigsOfType(whClass.NAME))
 
-    def AddActiveWarehouse(self, warehouseName, wcm: WarehouseClassManager) -> None:
-        """ Add warehouse to the preferences using a menu with the warehouse preset if available """
-
-        whClass = wcm.GetClassFromName(warehouseName + "Warehouse")
+    def AddActiveClass(self, ioClass, config_category: str) -> None:
         try:
-            preset = whClass.ConfigurationPreset()  # type: ignore
+            preset = ioClass.ConfigurationPreset()
 
             if preset.HasQuestions():
+
+                if config_category == KEY_ACTIVE_ENTITIES:
+                    # Ask for Tag if the entity allows multi-instance - multi-instance has sense only if a preset is available
+                    if ioClass.AllowMultiInstance():
+                        preset.AddTagQuestion()
+
                 self.DisplayMessage(messages.PRESET_RULES)
+                self.DisplayMessage(
+                    f"Configure {ioClass.NAME} {ioClass.CATEGORY_NAME}")

```

Comment:
`ioClass.CATEGORY_NAME` could be read from the name of the parent class, so we don't need another manual parameter.